src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -70,18 +70,21 @@ private static void BindProperty(PropertyInfo property, object propertyOwner, IC
 
         private static object BindType(Type type, object typeInstance, IConfiguration configuration)
         {
-            var configValue = configuration.Get(null);
+            string configValue = null;
+            if (configuration is IConfigurationSection)
+            {
+                configValue = ((IConfigurationSection)configuration).Value;
+            }
             var typeInfo = type.GetTypeInfo();
 
             if (configValue != null)
             {
                 // Leaf nodes are always reinitialized
-                return CreateValueFromConfiguration(type, configValue, configuration);
+                return CreateValueFromConfiguration(type, configValue, configuration as IConfigurationSection);
```

Comment:
Better yet, use as instead of is and then a cast, and just check for null.

Can we also rename this method?  I strongly dislike BindType as a method name